### DISCUSSION TITLE
Keyboard leds

### DISCUSSION THREAD

**Discussion by abaddon201** [hathach/tinyusb#1016] [Q&A]
Hello!
I'm trying to control keyboard leds from my pico pi, configured as host. I've changed example of the host, to send reports to the keyboard, like that: in 'process_kbd_report' I added this code:

`

    static void process_kbd_report(uint8_t dev_addr, uint8_t instance, hid_keyboard_report_t const *report) {

                    static uint8_t rpt2 = 1;

                    printf("\trpt='0x%x'\n", rpt2);

                    tuh_hid_set_report(dev_addr, instance, 0, 2, &rpt2, 1);

                    if(!rpt2) rpt2 = 1;

                    rpt2<<=1;

        // after this all original code to echo key character is going
        ...
    }
`
rpt2 - is a static uint8_t

This turns off and on leds, but in weird sequence, for example:
rpt2: result
1: CAPS
2: SCROLL
4: SCROLL+NUM
8: SCROLL
16: SCROLL
32: SCROLL+CAPS
64: NUM
128: NUM
0: CAPS
here is repeating of previous pattern, until 128
128: ALL OFF
and repeat again...

If I try to continue, it's continue for some time, and then turnes off all leds. Also, sometimes, I get panic in controller.
I can't understand what is going on. It's looks like provided data for the report is ignored, and it's report something from the memory...
Can someone advise me to the right direction?
Also I tried to use HID_OUTPUT macro, but it looks like it doesn't change behavior...

**Comment by hathach** [comment]
1. First, please post your modified code along with some of original one in code block. The code order is relevant as well 
2. have you tried to build with LOG=3 and look at usb log, if not please do so, and post your log here as attached txt. It will help to understand what the stack is doing.
